tests/losses/test_mkmmd_loss.py

```diff
@@ -206,13 +455,17 @@ def test_beta_with_largest_hat_d() -> None:
     assert torch.all(one_hot_beta_min.eq(beta_target_min))
 
 
-def test_optimize_betas_degenerate_case() -> None:
-    # The simple test cases above result in a set of hat_ds that are all negative. In this case, we perform the
-    # selection of a single kernel as recommended in Gretton
-    degenerate_betas = mkmmd_loss.optimize_betas(X, Y, lambda_m=0.0001)
-    beta_target = torch.Tensor([[0, 1, 0]]).to(DEVICE).t()
+def test_optimize_betas_for_X_Y() -> None:
+    # The simple test cases above result in a set of hat_ds that are all negative for the linear estimate. In this
+    # case, we perform the selection of a single kernel as recommended in Gretton
+    degenerate_betas = mkmmd_loss_linear.optimize_betas(X, Y, lambda_m=0.0001)
+    beta_target = torch.Tensor([[1, 0, 0]]).to(DEVICE).t()
     assert torch.all(degenerate_betas.eq(beta_target))
 
+    non_degenerate_beta = mkmmd_loss.optimize_betas(X, Y, lambda_m=0.0001)
+    beta_target = torch.Tensor([[0, 0, 1.0]]).to(DEVICE).t()
```

Comment:
it is funny that even here with and without linear approximation we have such a different optimized betas.

Reply:
Yeah, I'm thinking that the linear approximation is just bad. The full approximation values for the mean and variance make more sense to me.